### PR TITLE
fix: App context event handling for channel and membership

### PR DESCRIPTION
**Pull request by mohitpubnub**
fix: app context events handling.

Fixes issue of App context event handling for channel and membership.

### CODE REVIEW

File: src/event-engine/index.ts [hunk 1]
```diff
@@ -136,11 +136,11 @@ export class EventEngine {
   }
 
   getSubscribedChannels(): string[] {
-    return this.channels.slice(0);
+    return Array.from(new Set(this.channels.slice(0)));
   }
 
   getSubscribedChannelGroups(): string[] {
-    return this.groups.slice(0);
+    return Array.from(new Set(this.groups.slice(0)));
```
Comment (parfeon) [inline src/event-engine/index.ts:143]: Do we really need `slice(0)` if there is only primitive “string”?
```suggestion
    return Array.from(new Set(this.groups));
```
If there were an object or another array - then yup, we will need to make a “copy” of it.

File: src/event-engine/index.ts [hunk 2]
```diff
@@ -136,11 +136,11 @@ export class EventEngine {
   }
 
   getSubscribedChannels(): string[] {
-    return this.channels.slice(0);
+    return Array.from(new Set(this.channels.slice(0)));
```
[the remark this answers is not in the document]
Comment (mohitpubnub) [reply src/event-engine/index.ts:139]: Yes, I think we can go with reference passing 👍🏻

File: src/core/endpoints/subscriptionUtils/receiveMessages.js [hunk 3]
```diff
@@ -56,8 +56,9 @@ const endpoint = {
         issuingClientId: envelope.i,
         subscribeKey: envelope.k,
         originationTimetoken: envelope.o,
+        userMetadata: envelope.u,
         publishMetaData: {
-          timetoken: envelope.p.t,
```
Comment (parfeon) [inline src/core/endpoints/subscriptionUtils/receiveMessages.js:60]: Is there a reason to rename field? It is alsready inside of `publishMetaData` and it doesn't look like that we need to clarify once again that it is `publishTimetoken`. Is there other reason for it?

Comment (mohitpubnub) [reply src/core/endpoints/subscriptionUtils/receiveMessages.js:60]: It was typo before.
In eventEmitter (the same code was there in subscriptionManager before) while creating event information object, [this](https://github.com/pubnub/javascript/blob/26bb23c85943d54094139b4639736b344db26c63/src/core/components/eventEmitter.js#L69) was the way we used to assign.

so instead of chaning property name in all places (which may be error prone like we have now with channel/membership) I rename it here in subscription utility method.. same as older endpoint implementation [here](https://github.com/pubnub/javascript/blob/26bb23c85943d54094139b4639736b344db26c63/src/core/endpoints/subscribe.js#L64) 

in short, It's renaming to make is compatible with old implementation logic

Comment (mohitpubnub) [reply src/core/endpoints/subscriptionUtils/receiveMessages.js:60]: this can be refactored other way around than as it's there!

Comment (mohitpubnub) [reply src/core/endpoints/subscriptionUtils/receiveMessages.js:60]: @parfeon  refactored!

**Review by parfeon — APPROVED** [review]
LGTM!

**Comment by mohitpubnub** [comment]
@pubnub-release-bot release as v7.6.1

**Review by techwritermat — APPROVED** [review]
